### DISCUSSION TITLE
Rule on tele-switch1#state do publish stat/control/motion1 %value% endon

### DISCUSSION THREAD

**Discussion by miazza99** [arendst/Tasmota#21254] [Support request] [answered]
Hello,

I have an ESP32 I/O board that is flashed with the Tasmota FW.
I have 6 I/O port that are configured as switch (1 to 6) and an MQTT rule that triggers the switchs status change in order to publish a topic :
Rule1 on switch1#state do publish stat/control/motion1 %value% endon on switch2#state do publish stat/control/motion2 %value% endon ... and so on. for all the 6 switches.
This works fine and the rule is triggered at any switch status change.

Now I need that the switch status is published on regular basis in the same way independently from the switch change status in order to keep OpenHab refreshed on the status. 
I tried tu use the tele- in a rule as below:

Rule3 on tele-switch1#state do publish stat/control/motion1 %value% endon
Rule3 1

But this does not trigger in any way. I also tried to change the TelePeriod to 60 to speed up the try but the rule is nevere triggered.

What am I doing wrong ?

**Comment by sfromis** [comment]
Events with `#state` are **only** when it happens, not later, thus not relevant at TelePeriod.

Tasmota is **already** publishing Switch states at a regular basis, without you doing anything, in the `STATE` messages published at TelePeriod. And using `SetOption114`, you can have Tasmota publish Switch state changes as they happen. No rules needed.

However, you *can* create rules using the existing Switch state from the telemetry messages, like `on tele-Switch1=ON do ... endon. This is not the same as `#state`. 

Additionally, unrelated to TelePeriod, you can also use rule variables like `%switch1%` in any command, like Publish, executed when you want.

**Comment by miazza99** [comment]
Thank you sfromis. Now it is clear why I did not see any trigger.

Considering that at OH wake-up the switch state is unknown following your suggestion shall I make a rule like the one below for each of the switch ?
`on tele-switch1=ON do publish stat/control/motion1 %value% endon on tele-switch1=OFF do publish stat/control/motion1 %value% endon`

Or is there any clever way for having the same topic published from the OH command field ?

**Reply by sfromis** [reply]
My example was only to show how it works, including having text values instead of numbers. You need no compare in the trigger; if you are ok with the %value%, one is enough.

As IDK what you mean by "same topic", I have no answer for that. Never been near openHAB.

Maybe you'd want rules triggering on `Switch1#Boot` ...

But still, you need no rules at all, as Tasmota can already publish switch states, both as it happens, and at TelePeriod.

**Reply by miazza99** [reply]
OK. I will try. (sorry but as you have understood I am a very beginner in this stuff).
When I say _"same topic"_ I mean that in OH I need exactly the sintax `stat/control/motion1 %value%` and this is not the case at TelePeriod. This is why I started from triggering a rule.

**Reply by sfromis** [reply, marked as answer]
I do think that openHAB can manage to parse JSON data in the *payload*, without all that extra trouble.

**Reply by miazza99** [reply]
Finally the JSON  seems the best solution and without rules :)
I have set `SwitchMode1 15` and used the OH JSON `JSONPATH:$.Switch1` on the `tele SENSOR` topic and it worked at the first try !
Thank you for the assistance.
